src/App.tsx

```diff
@@ -14,11 +14,113 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+
+type State = {
+  selectedGood: string[],
+};
+
+class App extends React.Component<Props, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  messageSelectGood = () => {
+    const { selectedGood } = this.state;
+
+    if (this.state.selectedGood.length === 1) {
+      return `${selectedGood} is selected`;
+    }
+
+    if (this.state.selectedGood.length > 1) {
+      const firstWorlds = selectedGood.slice(0, selectedGood.length - 1);
+      const lastWorld = selectedGood[selectedGood.length - 1];
+
+      return `${firstWorlds.join(', ')} and ${lastWorld} are selected`;
+    }
+
+    return 'No goods selected';
+  };
+
+  addGood = (item: string) => {
+    this.setState((state) => {
+      return {
+        selectedGood: [
+          ...state.selectedGood,
+          item,
+        ],
+      };
+    });
+  };
+
+  removeGood = (item: string) => {
+    this.setState((state) => {
+      return {
+        selectedGood: state.selectedGood.filter(el => item !== el),
+      };
+    });
+  };
+
+  resetGood = () => {
+    this.setState({ selectedGood: [] });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>
+          Selected good: -
+          {' '}
+          {this.messageSelectGood()}
+        </h1>
+        <div className="block-button">
+          {(this.state.selectedGood.length < 1) || (
```

Comment:
```suggestion
          {(!this.state.selectedGood.length) && (
```
Використовуй && для conditional rendering